classes/Http/Controller/TalkFormController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Controller;
+
+use OpenCFP\Domain\CallForPapers;
+use OpenCFP\Domain\Model\Talk;
+use OpenCFP\Domain\Services\Authentication;
+use OpenCFP\Http\View\TalkHelper;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig_Environment;
+
+class TalkFormController extends BaseController
+{
+    private $authentication;
+    private $talkHelper;
+    private $callForPapers;
+
+    public function __construct(
+        Authentication $authentication,
+        TalkHelper $talkHelper,
+        CallForPapers $callForPapers,
+        Twig_Environment $twig,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->authentication = $authentication;
+        $this->talkHelper     = $talkHelper;
+        $this->callForPapers  = $callForPapers;
+
+        parent::__construct($twig, $urlGenerator);
+    }
+
+    public function editAction(Request $request)
+    {
+        $talkId = (int) $request->get('id');
+        // You can only edit talks while the CfP is open
+        // This will redirect to "view" the talk in a read-only template
+        if (!$this->callForPapers->isOpen()) {
+            $request->getSession()->set(
+                'flash',
+                [
+                    'type'  => 'error',
+                    'short' => 'Read Only',
+                    'ext'   => 'You cannot edit talks once the call for papers has ended', ]
+            );
+
+            return new RedirectResponse($this->url('talk_view', ['id' => $talkId]));
+        }
+
+        if (empty($talkId)) {
+            return $this->redirectTo('dashboard');
+        }
+
+        $userId = $this->authentication->userId();
+
+        $talk = Talk::find($talkId);
+
+        if (!$talk instanceof Talk || (int) $talk['user_id'] !== $userId) {
+            return $this->redirectTo('dashboard');
+        }
+
+        $data = [
```

Comment:
While we are at it, can we inline `$data`?

Also see #851, where I already started working on that.